### PR TITLE
Generate Dockerfile and container.yaml

### PR DESCRIPTION
**Pull request by lgarciaaco**
Generate `container.yaml` and `Dockerfile` as part of the `syndesis-operator olm` command. Those two files were missing but including them will ease the automation of this process.

### CODE REVIEW

File: install/operator/pkg/cmd/internal/olm/olm.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func New(parent *internal.Options) *cobra.Command {
 }
 
 func (o *Olm) generate() (err error) {
-	conf, err := configuration.GetProperties(context.TODO(), configuration.TemplateConfig, o.ClientTools(), &v1beta1.Syndesis{})
+	conf, err := configuration.GetProperties(context.TODO(), configuration.TemplateConfig, nil, &v1beta1.Syndesis{})
```
Comment (phantomjinx) [inline install/operator/pkg/cmd/internal/olm/olm.go:53]: Why is this necessary?

Is it to stop the ApiCapabilities being analysed?

File: install/operator/pkg/syndesis/configuration/configuration.go [hunk 2]
```diff
@@ -353,12 +354,12 @@ func GetProperties(ctx context.Context, file string, clientTools *clienttools.Cl
...
 
-	if ac, err := capabilities.ApiCapabilities(clientTools); err != nil {
-		return nil, err
-	} else {
-		configuration.ApiServer = *ac
+		if ac, err := capabilities.ApiCapabilities(clientTools); err != nil {
+			return nil, err
+		} else {
+			configuration.ApiServer = *ac
+		}
```
Comment (phantomjinx) [inline install/operator/pkg/syndesis/configuration/configuration.go:362]: Is there something causing problems with this?

File: install/operator/pkg/syndesis/olm/annotations_test.go [hunk 3]
```diff
@@ -36,4 +36,6 @@ func Test_annotation_build(t *testing.T) {
 	err = a.build()
 	assert.NoError(t, err)
 	assert.NotNil(t, a.body)
+
+	// fmt.Print(string(a.body))
```
Comment (phantomjinx) [inline install/operator/pkg/syndesis/olm/annotations_test.go:40]: Do you still want these?

File: install/operator/pkg/syndesis/olm/docker.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      summary="Operator which manages the lifecycle of the Fuse Online application." \
+      description="Operator which manages the lifecycle of the Fuse Online application." \
+      com.redhat.component="fuse-online-operator-metadata-container" \
+      io.k8s.description="Operator which manages the lifecycle of the Fuse Online application." \
+      io.k8s.display-name="Red Hat Fuse Online Operator" \
+      io.openshift.tags="fuse"
+`
+	m = fmt.Sprintf(m, channel, channel, c.config.SupportedOpenShiftVersions, c.config.Version)
+	c.body = []byte(m)
+	return
```
Comment (phantomjinx) [inline install/operator/pkg/syndesis/olm/docker.go:61]: Do you need this "return"?

File: install/operator/pkg/syndesis/olm/docker_test.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/syndesisio/syndesis/install/operator/pkg/apis/syndesis/v1beta1"
+	"github.com/syndesisio/syndesis/install/operator/pkg/syndesis/configuration"
+	syntesting "github.com/syndesisio/syndesis/install/operator/pkg/syndesis/testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_docker_build(t *testing.T) {
+	clientTools := syntesting.FakeClientTools()
+	conf, err := configuration.GetProperties(context.TODO(), "../../../build/conf/config-test.yaml", clientTools, &v1beta1.Syndesis{})
```
Comment (phantomjinx) [inline install/operator/pkg/syndesis/olm/docker_test.go:32]: You've included a clientTools in your _conf_ here but not used in that actual implementation. Probably makes little difference if d.build() doesn't use it but might be problematic in the future?
Might be safer for them to be consistent since the test would at least flag up the same result as the implementation.

File: install/operator/pkg/cmd/internal/olm/olm.go (hunk 1 above)
Comment (lgarciaaco) [reply install/operator/pkg/cmd/internal/olm/olm.go:53]: I don't need to any client in the context of the `olm` command. I think the best way to express this in a declarative way is setting the clientool to nil. The change below is to make this code work.

File: install/operator/pkg/syndesis/olm/docker_test.go (hunk 5 above)
Comment (lgarciaaco) [reply install/operator/pkg/syndesis/olm/docker_test.go:32]: You mean I dont need the clienttool and it is enough with nil?

File: install/operator/pkg/syndesis/olm/docker.go (hunk 4 above)
Comment (lgarciaaco) [reply install/operator/pkg/syndesis/olm/docker.go:61]: Yes, the function returns an error

File: install/operator/pkg/syndesis/configuration/configuration.go (hunk 2 above)
Comment (lgarciaaco) [reply install/operator/pkg/syndesis/configuration/configuration.go:362]: Yes. As commented above I don't need a client tool for `olm` command. this command will run in CI systems most likely. The block of code I put inside the condition throws an error if the clienttool is nil.

File: install/operator/pkg/syndesis/olm/annotations_test.go (hunk 3 above)
Comment (lgarciaaco) [reply install/operator/pkg/syndesis/olm/annotations_test.go:40]: I can remove

**Comment by lgarciaaco** [comment]
Fixes https://issues.redhat.com/browse/ENTESB-13607

**Comment by avano** [comment]
retest this please
